packages/fogbender-element/src/utils.ts

```diff
@@ -13,23 +15,35 @@ export const noopCleanup = () => {
   return new Promise<() => void>(resolve => resolve(() => {}));
 };
 
-export const renderIframe = (divRef: HTMLDivElement | undefined, headless: boolean) => {
-  const fogbender = useContext(FogbenderContext);
+export const renderIframe = (
+  divRef: HTMLDivElement | undefined,
+  headless: boolean,
+  element: ICustomElement
+) => {
+  const fogbender: Fogbender = consume(fogbenderContext, element as HTMLElement & ICustomElement);
 
-  createEffect(() => {
-    if (fogbender && divRef) {
-      renderComponent(() => fogbender().renderIframe({ headless, rootEl: divRef }));
-    } else {
-      return noopCleanup();
-    }
-  });
+  if (!fogbender) {
+    throw new Error("No fogbender set, use FogbenderProvider to set one");
+  }
+
+  if (fogbender && divRef) {
+    return renderComponent(() => fogbender.renderIframe({ headless, rootEl: divRef }));
+  } else {
+    return noopCleanup();
+  }
 };
 
 export const renderComponent = (componentRenderer: () => Promise<() => void>) => {
+  const [cleanup, setCleanup] = createSignal(() => {});
```

Comment:
does it have to be a signal?

Reply:
I just want this to be in controlled way that's why I created this as a signal.
Normal function declaration will also do the work.